var/spack/repos/builtin/packages/gmap-gsnap/package.py

```diff
@@ -37,3 +37,41 @@ class GmapGsnap(AutotoolsPackage):
     version('2018-02-12', '13152aedeef9ac66be915fc6bf6464f2')
     version('2017-06-16', 'fcc91b8bdd4bf12ae3124de0c00db0c0')
     version('2014-12-28', '1ab07819c9e5b5b8970716165ccaa7da')
+
+    variant(
+        'simd',
+        description='CPU support.',
+        values=('avx2', 'sse42', 'avx512', 'sse2'),
+        multi=True
+    )
+
+    def install(self, spec, prefix):
+        configure = Executable('../configure')
+
+        if 'simd=avx2' in self.spec:
+            with working_dir('avx2', create=True):
+                configure('--with-simd-level=avx2',
+                          '--prefix={0}'.format(prefix))
+                make()
+                make('install')
+
+        if 'simd=sse42' in self.spec:
+            with working_dir('sse42', create=True):
+                configure('--with-simd-level=sse42',
+                          '--prefix={0}'.format(prefix))
+                make()
+                make('install')
+
+        if 'simd=avx512' in self.spec:
+            with working_dir('avx512', create=True):
+                configure('--with-simd-level=avx512',
+                          '--prefix={0}'.format(prefix))
+                make()
+                make('install')
+
+        if 'simd=sse2' in self.spec:
+            with working_dir('sse2', create=True):
+                configure('--with-simd-level=sse2',
+                          '--prefix={0}'.format(prefix))
+                make()
+                make('install')
```

Comment:
I would rewrite things as follows:
```python
class GmapGsnap(AutotoolsPackage):
    ...
    def configure(self, spec, prefix):
        configure = Executable('../configure')

        for simd in spec.variants['simd'].value:
            with working_dir(simd, create=True):
                configure('--with-simd-level={0}'.format(simd),
                          '--prefix={0}'.format(prefix))

    def build(self, spec, prefix):
        for simd in spec.variants['simd'].value:
            with working_dir(simd):
                make()

    def check(self):
        for simd in self.spec.variants['simd'].value:
            with working_dir(simd):
                make('check')

    def install(self, spec, prefix):
        for simd in spec.variants['simd'].value:
            with working_dir(simd):
                make('install')
```
The benefit of this is that it keeps things in `AutotoolsPackage` and keeps `spack configure` and `spack build` functionality, as well as supports `spack install --test` (although you may need to check if the Makefile actually has a `check` or `test` target.

Reply:
Thank you!